### PR TITLE
T15574

### PR DESCRIPTION
**Pull request by smspillaz**
This branch makes everything ESLint complaint

https://phabricator.endlessm.com/T15574

### CODE REVIEW

File: lib/controller.js [hunk 1]
```diff
@@ -563,15 +570,15 @@ const CodingGameController = new Lang.Class({
...
-                    break;
-                default:
+            case 'add-app':
+                this._effectsManager.removeApplication(e.data.app);
+                break;
+            case 'remove-app':
+                this._effectsManager.addApplication(e.data.app);
+                break;
+            default:
                     // Crash here, shouldn't be reachable.
```
Comment (krnowak) [inline lib/controller.js]: You could unindent the comment too.

File: tests/js/testCodingGameServiceTimeline.js [hunk 2]
```diff
@@ -38,59 +38,59 @@ function walkTimelineForEventSequence(timeline) {
...
+                        subsequence: event.data.responses[response].map(addTriggerEvent)
                     });
-                    break;
-                case 'start-mission':
-                    let mission = Controller.findInArray(timeline.missions,
+                });
+                break;
+            case 'start-mission':
+                let mission = Controller.findInArray(timeline.missions,
                                                          m => m.name === event.data.name);
```
Comment (krnowak) [inline tests/js/testCodingGameServiceTimeline.js]: This seems to be off, now.

File: tests/js/testCodingGameServiceTimeline.js [hunk 3]
```diff
@@ -38,19 +37,20 @@ function walkTimelineForEventSequence(timeline) {
...
                     });
-                    break;
-                case 'start-mission':
+                });
+                break;
+            case 'start-mission':
+                (function() {
                     let mission = Controller.findInArray(timeline.missions,
-                                                         m => m.name === event.data.name);
+                                                             m => m.name === event.data.name);
```
Comment (krnowak) [inline tests/js/testCodingGameServiceTimeline.js]: This seems to be off, now.

File: lib/controller.js [hunk 4]
```diff
@@ -674,20 +681,20 @@ const CodingGameController = new Lang.Class({
...
-            this._dispatch({
-                name: event.name + '::completed',
-                type: 'wait-for-complete',
-                data: {
-                    name: event.name
-                }
-            });
-
-            let toDispatch = findEventsToDispatchInDescriptors(event.data.then,
+                                                this._dispatch({
```
Comment (krnowak) [inline lib/controller.js]: Would moving the lambda to a separate patch rein the indentation a bit?

File: mocks/constructors.js [hunk 5]
```diff
@@ -1,3 +1,4 @@
+/* eslint-env jasmine */
```
Comment (krnowak) [inline mocks/constructors.js]: Would be nice to explain in commit message about the reasons.

File: bin/edit-desktop-file.js [hunk 6]
```diff
@@ -27,7 +27,7 @@ let showHelp = (ARGV.length === 1 && ARGV[0] === 'help')
     || ARGV.indexOf('--help') !== -1;
 
 if (ARGV.length < 2 || showHelp) {
-    usage.map(line => print(line));
+    log(usage.join('\n'));
```
Comment (krnowak) [inline bin/edit-desktop-file.js]: What does `print` do? Maybe write it in the commit message.

File: bin/coding-game-service.js.in [hunk 7]
```diff
@@ -55,9 +57,11 @@ const Service = imports.lib.service;
 function loadTimelineDescriptors(cmdlineFile) {
     let filesToTry = [
         cmdlineFile,
-        Gio.File.new_for_path(GLib.build_filenamev([GLib.get_user_config_dir(),
-                                                    'com.endlessm.CodingGameService',
-                                                    'timeline.json'])),
+        Gio.File.new_for_path(GLib.build_filenamev([
```
Comment (krnowak) [inline bin/coding-game-service.js.in]: Heh, this indentation is crazy. It's different from the one used for lambdas. Feels like inconsistency.

File: bin/coding-game-service.js.in [hunk 8]
```diff
@@ -131,9 +131,11 @@ const CodingGameServiceApplication = new Lang.Class({
 
     vfunc_dbus_register: function(conn, object_path) {
         this.parent(conn, object_path);
-        let logFile = Gio.File.new_for_path(GLib.build_filenamev([GLib.get_user_config_dir(),
-                                                                  'com.endlessm.CodingGameService',
-                                                                  'game-service.log']));
+        let logFile = Gio.File.new_for_path(GLib.build_filenamev([
```
Comment (krnowak) [inline bin/coding-game-service.js.in]: That probably could be squashed with the `bin: Use correct indentation for multiline arrays` commit.

File: lib/controller.js [hunk 9]
```diff
@@ -43,20 +43,24 @@ function resolveGSettingsValue(value, oldValue) {
     if (typeof(value) === 'object') {
         switch(value.type) {
         case 'internal-file-uri':
-            let path = GLib.build_filenamev([
-                Config.coding_files_dir,
-                value.value
-            ]);
-            return GLib.filename_to_uri(path, null);
+            return (function() {
```
Comment (krnowak) [inline lib/controller.js]: Couldn't you just use, you know, brackets? To create an inner scope. :)

File: lib/desktopFile.js [hunk 10]
```diff
@@ -125,8 +123,8 @@ const Editor = new Lang.Class({
         this._path = userFullPath.get_path();
         this._metadataPath = this._path + '.json';
 
-        let [_, fullPath] = this._keyFile.load_from_data_dirs(path,
-                GLib.KeyFileFlags.KEEP_TRANSLATIONS);
+        let fullPath = this._keyFile.load_from_data_dirs(path,
+                GLib.KeyFileFlags.KEEP_TRANSLATIONS)[1];
```
Comment (krnowak) [inline lib/desktopFile.js]: Doesn't eslint complain about the indentation of the second parameter here?

File: lib/communicator.js [hunk 11]
```diff
@@ -181,9 +181,9 @@ const ExternalEffects = new Lang.Class({
         this._activeTimeouts[name] = GLib.timeout_add(GLib.PRIORITY_DEFAULT,
                                                       timeout,
                                                       Lang.bind(this, function() {
-            delete this._activeTimeouts[name];
-            callback();
-        }));
+                                                          delete this._activeTimeouts[name];
```
Comment (krnowak) [inline lib/communicator.js]: Maybe make lambda a separate variable to avoid crazy long indentation?

File: tests/js/testCodingGameServiceTimeline.js [hunk 12]
```diff
@@ -48,18 +48,20 @@ function walkTimelineForEventSequence(timeline) {
                 });
                 break;
             case 'start-mission':
-                let mission = Controller.findInArray(timeline.missions,
-                                                         m => m.name === event.data.name);
+                (function() {
```
Comment (krnowak) [inline tests/js/testCodingGameServiceTimeline.js]: Why not just use brackets?

**Comment by smspillaz** [comment]
> Isn't there any javascript language construct that tells "this variable is exported"? There are some many "eslint disable this warning" comment lines, which makes me wonder about the usefullness of the tool.

There is! Unfortunately it isn't supported on Gjs and ESLint is a little node-centric

File: lib/controller.js (hunk 9 above)
Comment (smspillaz) [inline lib/controller.js]: Hah! Of course, didn't actually cross my mind :p

**Comment by ptomato** [comment]
That's too bad, jshint had a special `/* exported variable1, variable2 */` comment you could add at the top of a file. I suppose you could do `void variable1, variable2;` at the bottom of each file if you didn't want to have individual comments on each line that contained an exported variable.

File: bin/coding-game-service.js.in (hunk 7 above)
Comment (smspillaz) [inline bin/coding-game-service.js.in]: I don't think its that bad - hanging brackets is how things are done for lists generally

File: lib/desktopFile.js (hunk 10 above)
Comment (smspillaz) [inline lib/desktopFile.js]: Unfortunately not :( There actually doesn't seem to be any default rules about vertical alignment of parameters, so ESLint kind of just lets them go anywhere. We could write a plugin though.

File: lib/communicator.js (hunk 11 above)
Comment (smspillaz) [inline lib/communicator.js]: Yes, of course

**Comment by smspillaz** [comment]
> That's too bad, jshint had a special /* exported variable1, variable2 */ comment you could add at the top of a file. I suppose you could do void variable1, variable2; at the bottom of each file if you didn't want to have individual comments on each line that contained an exported variable.

There might also be some scope to write a plugin to deal with this problem - in the same vein as `/* exported */` . Another option would be to define an empty object called `module` in Gjs and then use `module.exports = { ... }` - that way ESLint will stop complaining about it and we also get checking to make sure that we're not tagging anything as exported that doesn't exist.

File: lib/log.js [hunk 13]
```diff
@@ -48,11 +51,13 @@ const CodingGameServiceLog = new Lang.Class({
         try {
             let logFileDirectory = this._logFile.get_parent();
             logFileDirectory.make_directory_with_parents(null);
-        } catch (e if e.matches(Gio.IOErrorEnum, Gio.IOErrorEnum.EXISTS)) {
-            // Ignore this case
         } catch (e) {
-            logError(e, 'Unable to create game service log directory');
-            // Keep going -- we'll fail anyway
+            if (e.matches(Gio.IOErrorEnum, Gio.IOErrorEnum.EXISTS)) {
+            // Ignore this case
```
Comment (krnowak) [inline lib/log.js]: Indent the comment.

File: lib/log.js [hunk 14]
```diff
@@ -209,9 +215,9 @@ const CodingGameServiceLog = new Lang.Class({
     reversedCopyFileEvents: function() {
       // Go through the event log in reverse and yield copy-file
```
Comment (krnowak) [inline lib/log.js]: You could fix the indentation of these comments too, like you did for the code just below them.

File: bin/edit-desktop-file.js [hunk 15]
```diff
@@ -1,11 +1,12 @@
 #!/usr/bin/env gjs
-/* edit-desktop-file.js
- *
- * Copyright (c) 2016 Endless Mobile Inc.
- * All Rights Reserved.
- *
- * Simple script to exercise the desktop file editor.
- */
+/* global print */
```
Comment (krnowak) [inline bin/edit-desktop-file.js]: Does this need to be a `/*` comment (instead of the `//` comment)?

File: mocks/constructors.js (hunk 5 above)
Comment (krnowak) [inline mocks/constructors.js]: Does this need to be `/*` instead of `//`?

File: tests/js/testCodingGameService.js [hunk 16]
```diff
@@ -1,3 +1,4 @@
+/* eslint-env jasmine */
```
Comment (krnowak) [inline tests/js/testCodingGameService.js]: Does this need to be `/*` instead of `//`? (Just remembered that @cosimoc was nitpicking on this style of comments.)

File: tests/js/testCodingGameServiceTimeline.js [hunk 17]
```diff
@@ -1,3 +1,4 @@
+/* eslint-env jasmine */
```
Comment (krnowak) [inline tests/js/testCodingGameServiceTimeline.js]: Does this need to be `/*` instead of `//`?

File: lib/log.js (hunk 14 above)
Comment (smspillaz) [inline lib/log.js]: Yeah .... unfortunately the auto fix mode doesn't touch comments :( I'll fix these now.

File: bin/edit-desktop-file.js (hunk 15 above)
Comment (smspillaz) [inline bin/edit-desktop-file.js]: It would appear so.

File: tests/js/testCodingGameServiceTimeline.js [hunk 18]
```diff
@@ -133,62 +134,62 @@ describe('Default Game Service Controller Timeline', function () {
...
+        case 'wait-for':
+            describe('would wait ' + sequencePoint.timeout + ' to satisfy ' + sequencePoint.name, function() {
+                sequencePoint.subsequence.forEach(function(subsequencePoint) {
+                    subsequencePoint.forEach(sequencePointTestCaseGenerator);
+                });
+            });
+            break;
+        default:
+            throw new Error('Don\'t know how to handle sequence ' +
                                 'point type ' + sequencePoint.type);
```
Comment (cosimoc) [inline tests/js/testCodingGameServiceTimeline.js]: This line should be re-indented too

File: tests/js/testCodingGameServiceTimeline.js [hunk 19]
```diff
@@ -133,62 +134,62 @@ describe('Default Game Service Controller Timeline', function () {
...
-                describe(sequencePoint.event + ' received response ' + sequencePoint.response, function() {
-                    beforeAll(function() {
-                        controller.receiveChatResponse(sequencePoint.event,
+            });
+            break;
+        case 'chat-response':
+            describe(sequencePoint.event + ' received response ' + sequencePoint.response, function() {
+                beforeAll(function() {
+                    controller.receiveChatResponse(sequencePoint.event,
                                                        'A response',
```
Comment (cosimoc) [inline tests/js/testCodingGameServiceTimeline.js]: Reindent these lines too
